commands.go

```diff
@@ -28,8 +28,9 @@ const (
 	deleteUserMessageType = messaging.MessageType(0x32)
 
 	// Shard messages
-	createShardGroupIfNotExistsMessageType = messaging.MessageType(0x40)
-	deleteShardGroupMessageType            = messaging.MessageType(0x41)
+	createShardGroupIfNotExistsMessageType    = messaging.MessageType(0x40)
```

Comment:
They take different signatures.  The first shard creation happens the first time you write data.  Until that point in time, we don't know what to create.

After that, we pre-create based on existing shards.  So we have to have some pre-processing to determine time buckets for each one, then once that is known, it shares the same code.